pkg/controller/clusters/management.go

```diff
@@ -0,0 +1,45 @@
+package clusters
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+// FetchManagementEksaCluster returns the management cluster object for a given workload Cluster.
+// If we are unable to find the management cluster using the same namespace as the current cluster, we will attempt
+// to get a list of the clusters with that name across all the namespaces. If we find multiple, which usually should
+// not happen as these clusters get mapped to a cluster-api cluster object in the eksa-system namespace, then we
+// also error on that because it is not possible to have multiple resources with the same name within a namespace.
+func FetchManagementEksaCluster(ctx context.Context, cli client.Client, cluster *v1alpha1.Cluster) (*v1alpha1.Cluster, error) {
```

Comment:
I think we should write a test for this method
I'm ok doing it as a follow up if we need to get this is soon :)